### PR TITLE
Add the ability to have activation config secrets in a separate file

### PR DESCRIPTION
**Pull request by tomas-s-roos**
The `dt-sdk run` command now has an extra flag `--secrets`, which by default will be set to `secrets.json`.

This flag should receive the path to a JSON file which is then used to replace all instances of `{{key}}` in `activation.json` with `value`, for every `key`-`value` pair specified in the secrets file.

The `dt-sdk create` command now creates the `secrets.json` file by default and the template `activation.json` was adjusted to have an example. `secrets.json` is also now part of the default `.gitignore` file.

Updated the guide docs to have a short explanation of this new file.

A notable change is that the `DebugClient` class now requires an extra positional argument to be instantiated. Had to adjust some tests to ensure they didn't break, but it doesn't seem like this breaks anything else. Still, not sure if there's other code in the wild using `DebugClient` directly, which would be a problem.

### CODE REVIEW

File: dynatrace_extension/cli/main.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def version():
 def run(
     extension_dir: Path = typer.Argument("."),
     activation_config: str = "activation.json",
+    secrets: str = "secrets.json",
```
Comment (dlopes7) [inline dynatrace_extension/cli/main.py:43]: @tomas-s-roos this needs to be an optional argument so we don't break contract

File: dynatrace_extension/sdk/communication.py [hunk 2]
```diff
@@ -322,16 +322,26 @@ class DebugClient(CommunicationClient):
     def __init__(
         self,
         activation_config_path: str,
+        secrets_path: str,
```
Comment (dlopes7) [inline dynatrace_extension/sdk/communication.py:325]: Make this optional please so we don't make a breaking change here

File: tests/sdk/test_status.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def callback():
         ext = Extension()
         ext.logger = MagicMock()
         ext._running_in_sim = True
-        ext._client = DebugClient("", "", MagicMock())
+        ext._client = DebugClient("", "", "", MagicMock())
```
Comment (dlopes7) [inline tests/sdk/test_status.py:28]: When secrets is optional this should not change

**Review by dlopes7 — CHANGES_REQUESTED** [review]
Please make it optional everywhere

File: dynatrace_extension/cli/main.py (hunk 1 above)
Comment (tomas-s-roos) [reply dynatrace_extension/cli/main.py:43]: Here this is already optional, or am I missing something? It has a default value.

File: dynatrace_extension/sdk/communication.py (hunk 2 above)
Comment (tomas-s-roos) [reply dynatrace_extension/sdk/communication.py]: Fixed in latest PR

File: tests/sdk/test_status.py (hunk 3 above)
Comment (tomas-s-roos) [reply tests/sdk/test_status.py]: Fixed in latest commit

**Review by dlopes7 — APPROVED** [review]

**Pull request merged**